instrumentation-core/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/TracerAutoConfigurationTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import io.opentelemetry.OpenTelemetry;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+
+public class TracerAutoConfigurationTest {
+  private final ApplicationContextRunner contextRunner =
+      new ApplicationContextRunner()
+          .withConfiguration(AutoConfigurations.of(TracerAutoConfiguration.class));
+
+  @Test
+  public void should_contain_otel_tracer_bean() {
+    this.contextRunner.run(
+        (context) -> {
+          assertTrue(
+              "Application Context contains otelTracer bean", context.containsBean("otelTracer"));
+        });
+  }
+
+  @Test
+  public void should_set_otel_tracer_name() {
+    this.contextRunner
+        .withPropertyValues("opentelemetry.trace.tracer=testTracer")
+        .run(
+            (context) -> {
+              assertNotNull(
```

Comment:
[Note] Not needed since this is not internal Google code, but within Google we've moved to using Truth almost exclusively

It offers a much more fluent and intuitive style that is closer to other languages. e.g,
instead of a confusion like assertEquals(x,y)
you have something like
assertThat(x)isEqualTo(y).

Take a look some time at it: 
https://github.com/google/truth